specs/opcode/16AND.md

```diff
@@ -0,0 +1,50 @@
+# AND op code
+
+## Procedure  
+
+A stack initalize empty with stack pointer to 1024, AND operation can only happen when stack is not empty, and it will increase by 1 of stack pointer.  
+
+The And value will be dropped directly without any further checking & utilizing.
+
+## Constraints
+
+1. opId = OpcodeId(0x16u8)
+2. state transition:  
+    - gc + 3
+    - stack_pointer + 1
+    - pc + 1
+    - gas + 3
+
```

Comment:
Would like to see more descriptions of how the opcode is checked in the circuit so that the reviewers/auditors know better how to review the circuit code. For example, 
"We have two inputs `a` and `b` and an output `c`, all are EVM words. We break the EVM word into 32 bytes, lookup an AND table of 1-byte inputs, and apply the lookup to the 32 chunks of each `a`, `b`, and `c` to see if a[i] | b[i] === c[i] holds for i in 0~31"

Reply:
Add more descriptions